test/framework/options.py

```diff
@@ -2968,6 +2968,37 @@ def test_list_software(self):
         self.assertTrue(re.search('^\s+\* GCC v4.6.3: dummy', txt, re.M))
         self.assertFalse(re.search('gzip', txt, re.M))
 
+    def test_parse_optarch(self):
+        """Test correct parsing of optarch option."""
+        
+        options = EasyBuildOptions()
+
+        # Check for EasyBuildErrors
+        error_msg = "The optarch option has an incorrect syntax"
+        options.options.optarch = 'Intel:something;GCC'
+        self.assertErrorRegex(EasyBuildError, error_msg, options.postprocess)
+
+        error_msg = "The optarch option contains duplicated entries for compiler"
+        options.options.optarch = 'Intel:something;GCC:somethingelse;Intel:anothersomething'
+        self.assertErrorRegex(EasyBuildError, error_msg, options.postprocess)
+
+        # Check the parsing itself
+        test_cases = [
+                ('',''),
+                ('xHost','xHost'),
+                ('GENERIC','GENERIC'),
+                ('Intel:xHost', {'Intel': 'xHost'}),
+                ('Intel:GENERIC', {'Intel': 'GENERIC'}),
+                ('Intel:xHost;GCC:march=x86-64 -mtune=generic', {'Intel': 'xHost', 'GCC': 'march=x86-64 -mtune=generic'}),
+                ('Intel:;GCC:march=x86-64 -mtune=generic', {'Intel': '', 'GCC': 'march=x86-64 -mtune=generic'}),
+                ]
+
+        for optarch_string, optarch_parsed in test_cases:
+            options.options.optarch = optarch_string
+            options.postprocess()
+            self.assertTrue(options.options.optarch == optarch_parsed, "Parsing of optarch: True means %s == %s" % 
```

Comment:
Using `assertTrue` with `==` is a bit weird, so we should change that (but also in a separate PR).